task/backend/scheduler.go

```diff
@@ -470,19 +475,9 @@ func newTaskScheduler(
 	if err != nil {
 		return nil, err
 	}
-	maxC := defaultConcurrency
-
-	// if an environment variable for default concurrency is set, use this value
-	// this will be overwritten if the Flux script for the task has a concurrency set
-	if envConcurrency := os.Getenv("DEFAULT_CONCURRENCY"); envConcurrency != "" {
-		c, err := strconv.Atoi(envConcurrency)
-		if err == nil {
-			maxC = c
-		}
-	}
 
 	if opt.Concurrency != nil {
-		maxC = int(*opt.Concurrency)
```

Comment:
You should no mutate the `TickScheduler.maxConcurrency` field, as a single task that defines a concurrency option will update the default concurrency for all other tasks.